docs/01-app/05-api-reference/03-file-conventions/dynamic-routes.mdx

```diff
@@ -110,31 +68,46 @@ The difference between **catch-all** and **optional catch-all** segments is that
 
+| Route                               | `params` Type Definition                 |
+| ----------------------------------- | ---------------------------------------- |
+| `app/blog/[slug]/page.js`           | `{ slug: string }`                       |
+| `app/shop/[...slug]/page.js`        | `{ slug: string[] }`                     |
+| `app/shop/[[...slug]]/page.js`      | `{ slug?: string[] }`                    |
+| `app/[categoryId]/[itemId]/page.js` | `{ categoryId: string, itemId: string }` |
+
+## Behavior
+
+- Since the `params` prop is a promise. You must use `async`/`await` or React's use function to access the values.
+  - In version 14 and earlier, `params` was a synchronous prop. To help with backwards compatibility, you can still access it synchronously in Next.js 15, but this behavior will be deprecated in the future.
+
+## Examples
+
+### With `generateStaticParams`
+
+The [`generateStaticParams`](/docs/app/api-reference/functions/generate-static-params) function can be used to [statically generate](/docs/app/getting-started/partial-prerendering#static-rendering) routes at build time instead of on-demand at request time.
+
 ```tsx filename="app/blog/[slug]/page.tsx" switcher
-export default async function Page({
-  params,
-}: {
-  params: Promise<{ slug: string }>
-}) {
-  return <h1>My Page</h1>
+export async function generateStaticParams() {
+  const posts = await fetch('https://.../posts').then((res) => res.json())
+
+  return posts.map((post) => ({
+    slug: post.slug,
+  }))
 }
 ```
 
 ```jsx filename="app/blog/[slug]/page.js" switcher
-export default async function Page({ params }) {
-  return <h1>My Page</h1>
+export async function generateStaticParams() {
+  const posts = await fetch('https://.../posts').then((res) => res.json())
+
+  return posts.map((post) => ({
+    slug: post.slug,
+  }))
 }
 ```
 
-| Route                               | `params` Type Definition                 |
-| ----------------------------------- | ---------------------------------------- |
-| `app/blog/[slug]/page.js`           | `{ slug: string }`                       |
-| `app/shop/[...slug]/page.js`        | `{ slug: string[] }`                     |
-| `app/shop/[[...slug]]/page.js`      | `{ slug?: string[] }`                    |
-| `app/[categoryId]/[itemId]/page.js` | `{ categoryId: string, itemId: string }` |
-
-> **Good to know**: This may be done automatically by the [TypeScript plugin](/docs/app/api-reference/config/typescript#ide-plugin) in the future.
+The primary benefit of the `generateStaticParams` function is its smart retrieval of data. If content is fetched within the `generateStaticParams` function using a `fetch` request, the requests are [automatically deduplicated](/docs/app/deep-dive/caching#request-memoization). This means a `fetch` request with the same arguments across multiple `generateStaticParams`, Layouts, and Pages will only be made once, which decreases build times.
```

Comment:
I remember having a discussion about this very statement. I think it is correct, but just for x-ref, https://github.com/vercel/next.js/discussions/73959

Reply:
I glazed over this when reorganizing the content, but rereading it. I think it's a poorly worded comment, we should not be so definitive. It's really talking about deduplication, so I updated the comment to focus on that.